tests/unit/test_charm.py

```diff
@@ -42,36 +50,121 @@ def test_action_failures(self):
         )
 
     def test_config_changed(self):
-        self.harness.update_config({"database": "foo"})
+        self.harness.set_leader(True)
+        self.harness.update_config({"database-name": "foo"})
+        self.harness.charm._on_config_changed(Mock())
+        self.assertEqual(
+            self.harness.model.unit.status,
+            BlockedStatus("Please relate the data-integrator with the desired product"),
```

Comment:
I suppose this would miss the point of testing that the "right" blocked message is created. We have different block status (depending on the reason). We want to tests we get the one we want I suppose

Reply:
I will follow the suggestion done by @deusebio